view_user_birthdate.py

```diff
@@ -0,0 +1,242 @@
+"""Build a dictionary of usernames and their corresponding birthdates
+formatted to match their respective nationalities. Additionally, take
+in user input, match it to any usernames in the dictionary and return
+that user's formatted birthdate.
+"""
+import re
+from get_input import get_input
+
+class User(object):
+    """ Take in two strings 'username' and 'nationality' and a Date
+    'birthdate'. Create an object of type User with those same
+    variables as attributes. Keep track of all instantiated User
+    objects with the dictionary 'bday_book'.
+    """
+    bday_book = {}
+
+    def __init__(self, username, nationality, birthdate):
+        """Instantiate an object with the attributes 'username',
+        'nationality', and 'birthdate'. If user is not in the
+        bday_book, add the User's username and birthdate.
+        """
+        assert re.search(r"[\w.-]+", username), "Invalid username."
+        assert re.search("[a-zA-Z]+ ?[a-zA-Z]*", nationality), "Invalid nationality."
+        assert re.search("[0-9]{1,4}/[0-9]{1,2}/[0-9]{1,4}", birthdate)
+        self.username = username.lower()
+        self.nationality = nationality.lower()
+        self.birthdate = birthdate
+        # Add user to the bday_book.
+        if self.username not in User.bday_book:
+            User.bday_book[username] = self.birthdate
+
+class Date(object):
+    """Take in three strings 'month', 'day', and 'year' and create a
+    Date object with those same strings as the attributes.
+    """
+    def __init__(self, month, day, year):
+        """Instantiate a Date with the attributes 'month', 'day', and
+        'year'.
+        """
+        assert valid_month(make_int(month)), "Invalid month."
+        assert valid_day(make_int(day), make_int(month)), "Invalid day."
+        assert valid_year(make_int(year)), "Invalid year."
+        self.month = month
+        self.day = day
+        self.year = year
+
+class AmericanDate(Date):
+    """Create an object of type AmericanDate wich inherits from class
+    Date.
+    """
+    nationalities = ['american', 'belizean', 'micronesian']
+    def format(self):
+        """Using the attributes 'month', 'day', and 'year', return a
+        string in the American format:
+        mm/dd/yyyy
+        """
+        return '%s/%s/%s' % (self.month, self.day, self.year)
+
+class EuropeanDate(Date):
+    """Create an object of type EuropeanDate which inherits from class
```

Comment:
docs are too wordy here. this isn't telling me anything the code is not - tell me how eurpean dates differ from other dates?